### DISCUSSION TITLE
Remote ui testing similar to Chromatic

### DISCUSSION THREAD

**Discussion by jeromecovington** [storybookjs/react-native#411] [Q&A] [answered]
I have Storybook running as a view in my React Native iOS/Android app, and now it seems the next step would be to get a workflow going similar to what I am doing in my NextJS webapp with Storybook and Chromatic. I can't seem to find a happy path for such a workflow, but maybe this exists already? I was referred to https://github.com/storybookjs/native as a possibility for remote ui development and acceptance, but uploading the app to Appetize for each platform seems like a step backward and added complication since a React Native app is already by definition closer to the React and Storybook ecosystems than a "purely native" app would be. Any suggestions are greatly appreciated!

**Comment by dannyhw** [comment, marked as answer]
You could try a tool like owl or detox and there is a tool called loki. Although its not fully automatic like chromatic with some work it would be possible.

I plan to experiment with a few things once i have time but it should be possible also in the beta to use the main.js to create a list of stories and iterate through them and then take screenshots of each one with something like owl.

If you aren't worried about testing on a device you can also use addon-react-native-web and just use chromatic directly.

**Reply by dannyhw** [reply]
A new player in the testing space is maestro which has that sort of CI as a service thing, I haven't looked into it yet but I think it has potential.

Though I think a tight integration with something like owl might be closer to what chromatic offers.
